composition-js/src/hints.ts

```diff
@@ -235,6 +235,7 @@ export class CompositionHint {
   constructor(
     readonly definition: HintCodeDefinition,
     readonly message: string,
+    readonly coordinate: string | undefined,
```

Comment:
I would suggest changing this argument to be the (main) element that is the target of the hint, so something like:
```typescript
export class CompositionHint {
  public readonly nodes?: readonly SubgraphASTNode[];
  public readonly coordinate?: string;
  
  constructor(
    readonly definition: HintCodeDefinition,
    readonly message: string,
    element: NamedSchemaElement | undefined,
  ) {
    this.coordinate = element?.coordinate;
    ....
  }
  ...
}
```
Imo this has a bunch of advantages:
- this is more precisely typed. `string` is not the most helpful type: it's really easy to pass something that isn't a valid coordinate and have this go undetected. Passing `NamedSchemaElement` makes it less error prone and guaranteed we will get a valid coordinate at the end of the day.
- practically, this avoid having to repeat `.coodinate` on almost all callsite, which feel kinda nice.

Reply:
Thanks for the review! This makes sense to me, I will make this change and post a new revision